tools/export/exporters.py

```diff
@@ -159,3 +159,9 @@ def group_project_files(self, sources):
         """
         data = sorted(sources, key=self.make_key)
         return {k: list(g) for k,g in groupby(data, self.make_key)}
+
+    @abstractmethod
+    def generate(self):
+        """Generate an IDE/tool specific project file"""
+        raise NotImplemented("Implement a generate function in Exporter child class")
```

Comment:
Normally we shouldn't be changing code in documentation PRs if possible, but this seems simple enough to not be an issue. @sarahmarshy Are you aware of any way this could break existing Exporter child classes.?